src/Core/Services/StateMigrationService.cs

```diff
@@ -314,6 +317,90 @@ private async Task MigrateFrom2To3Async()
             await RemoveValueAsync(Storage.LiteDb, V2Keys.Keys_PassGenHistory);
         }
 
+        // v3 to v4 Migration
+
+        private class V3Keys
+        {
+            internal static string ThemeKey(string userId) => $"theme_{userId}";
+            internal static string AutoDarkThemeKey(string userId) => $"autoDarkTheme_{userId}";
+            internal static string DisableFaviconKey(string userId) => $"disableFavicon_{userId}";
+        }
+
+        private async Task MigrateFrom3To4Async()
+        {
+            var state = await GetValueAsync<State>(Storage.LiteDb, Constants.StateKey);
+            if (state?.Accounts != null)
+            {
+                string firstUserId = null;
+
+                // move values from state to standalone values in LiteDB
+                foreach (var account in state.Accounts)
+                {
+                    var userId = account.Value?.Profile?.UserId;
+                    if (userId != null)
+                    {
```

Comment:
You can write this as:
```c#
foreach (var account in state.Accounts.Where(a => account.Value?.Profile?.UserId != null))
```
and you avoid the if afterwards.
Also, if you were to keep the `if`, I would write it inverted so that it doesn't create arrow code